### PR TITLE
apis/cilium.io: do not regenerate deepcopy for unnecessary structs

### PR DESCRIPTION
**Pull request by aanm**
Fixes: 99f0136e65cb ("k8s: Restructure and optimize CEP")
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7030)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: pkg/k8s/apis/cilium.io/v2/types.go [hunk 1]
```diff
@@ -333,6 +333,7 @@ type EndpointStatus struct {
 const EndpointStatusLogEntries = 5
 
 // ControllerList is a list of ControllerStatus
+// +k8s:deepcopy-gen=false
```
Comment (ianvernon) [inline pkg/k8s/apis/cilium.io/v2/types.go:336]: What makes these structs "unnecessary"? Update the commit to provide more context regarding this.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Please provide more context for why deepcopy does not need to be generated for said types.

File: pkg/k8s/apis/cilium.io/v2/types.go (hunk 1 above)
Comment (aanm) [reply pkg/k8s/apis/cilium.io/v2/types.go:336]: Done

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Pull request merged**